server/models/chain.ts

```diff
@@ -46,20 +47,20 @@ export interface ChainAttributes {
   ChainObjectVersion?; // TODO
 }
 
-export interface ChainInstance extends Sequelize.Instance<ChainAttributes>, ChainAttributes {
+export interface ChainInstance extends Model<ChainAttributes>, ChainAttributes {
   // add mixins as needed
   getChainNodes: Sequelize.HasManyGetAssociationsMixin<ChainNodeInstance>;
 }
 
-export interface ChainModel extends Sequelize.Model<ChainInstance, ChainAttributes> {
-
-}
+type ChainModelStatic = typeof Model
```

Comment:
I wonder if this can be abstracted into a higher order typescript function + if there's an overall better design for these model files now that sequelize has native typescript support (starting from v5).

Reply:
something like
```
type ModelStatic<T extends Model> = typeof Model
    & { associate: (models: any) => void }
    & { new(values?: Record<string, unknown>, options?: BuildOptions): T }

type AddressModelStatic = ModelStatic<AddressInstance> & AddressCreationAttributes
```
?

Reply:
Yeah, if that works!

Reply:
that works !